client/src/components/ProtectedRoute/ProtectedRoute.jsx

```diff
@@ -0,0 +1,19 @@
+import { Route, Redirect } from 'react-router-dom';
+import { useAuth } from '../../context/useAuthContext';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+// eslint-disable-next-line
+const ProtectedRoute = ({ component, ...rest }) => {
```

Comment:
I believe you can simply write Component and use that directly instead of creating a separate Component variable.